src/h5web/storage-utils.ts

```diff
@@ -0,0 +1,54 @@
+import create, { State, StateCreator, StoreApi, UseStore } from 'zustand';
+
+export interface StorageConfig {
+  key: string;
+  whitelist: string[];
+}
+
+function fromStorage<T>(key: string): T | undefined {
+  try {
+    const value = window.localStorage.getItem(key);
+    return value !== undefined && value !== null
+      ? JSON.parse(value)
+      : undefined;
+  } catch {
+    // Access to local storage may not be authorized or available,
+    // or the data stored may be corrupted.
+    return undefined;
+  }
+}
```

Comment:
The naming puzzles me a bit here. I have the feeling that we get a single value from its key (ex: `colorMap`).

Perhaps I am simply not used to this so if it is the convention, leave it as it is.

Reply:
You can only store strings in local storage, but the string can be any valid JSON (a string, a number, a stringified object, etc.), so this function is meant as a generic utility to retrieve and parse any storage value. Would `readFromStorage` be any clearer?

Reply:
The function name is fine. What I mean is I am a bit confused as I see fields in the configuration as `(key, value)` pairs while here the key is the identifier of the config and the value the config itself.
In other words, one cannot do `fromStorage(colorMap)` to get the colorMap value. You have to get the whole config from its key/identifier, parse it and get the colorMap from it.

But that is not a problem, the naming stems from the fact that you want this function to be generic so you can leave it.